### PR TITLE
fix issue 15789

### PR DESCRIPTION
**Pull request by sobolewsk**
## Brief Information

This pull request is in the type of:

- [X] bug fixing
- [ ] new feature
- [ ] others

### What does this PR do?

Fixes https://github.com/apache/echarts/issues/15789

### Fixed issues

- 15789

### Before: What was the problem?

The `info` field (user defined data) was not available in the event handler.

### After: How does it behave after the fixing?

The `info` field is now properly passed to the event handlers. To verify please use the `test/custom-feature.html` test.

## Document Info

One of the following should be checked.

- [X] This PR doesn't relate to document changes
- [ ] The document should be updated later
- [ ] The document changes have been made in apache/echarts-doc#xxx

### ZRender Changes

- [ ] This PR depends on ZRender changes (ecomfe/zrender#xxx).

### Related test cases or examples to use the new APIs

N.A.

### Merging options

- [ ] Please squash the commits into a single one when merging.

### CODE REVIEW

**Review by Ovilia — CHANGES_REQUESTED** [review]
Thanks for your contribution! Please [add a test case](https://github.com/apache/echarts/wiki/How-to-make-a-pull-request#test).

**Comment by sobolewsk** [comment]
> 

@Ovilia it seems that the test case is already there: https://github.com/sobolewsk/echarts/blob/branches/fix-issue-15789/test/custom-feature.html#L124
However, it seems to be a manual test and I was having a problem trying to figure out how to test interactions like that in your Jest environment. Could you please point me to an example of a unit test which triggers click events?

**Comment by sobolewsk** [comment]
NVM, I figured it out. Unit test added.

File: test/ut/spec/series/custom.test.ts [hunk 1]
```diff
@@ -69,4 +69,46 @@ describe('custom_series', function () {
         expect(resultPaletteColors).toEqual(colors);
     });
 
+    it('should pass user defined data to event handlers', async () => {
+        const data = [
+            [10, 16,],
```
Comment (Ovilia) [inline test/ut/spec/series/custom.test.ts:74]: Extra comma here.

**Review by Ovilia — CHANGES_REQUESTED** [review]
Thanks for your contribution. Only a small problem needs to be fixed.

**Review by Ovilia — APPROVED** [review]

**Comment by sobolewsk** [comment]
Thank you @Ovilia for reviewing and merging!

**Pull request merged**